### ISSUE TITLE
Ubuntu 24.04 ansible-lint crashes due to _posix_flavour being removed

### ISSUE THREAD

**Issue by Koenvh1** [ansible/ansible-lint#4183]
##### Summary

![Screenshot from 2024-05-23 12-32-21](https://github.com/ansible/ansible-lint/assets/5168825/15636e89-1a5e-4016-8eee-a54eb11a267f)

The attribute `pathlib.PurePosixPath._posix_flavour` not longer exists in Python 3.12, as is used by wcmatch. I think it has been replaced with `pathlib.PurePosixPath._flavour`. This causes ansible-lint to crash on pretty much every command executed using the cli.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```
koen@beta:~$ ansible-lint --version
Traceback (most recent call last):
  File "/usr/bin/ansible-lint", line 5, in <module>
    from ansiblelint.__main__ import _run_cli_entrypoint
  File "/usr/lib/python3/dist-packages/ansiblelint/__main__.py", line 39, in <module>
    from ansiblelint import cli
  File "/usr/lib/python3/dist-packages/ansiblelint/cli.py", line 20, in <module>
    from ansiblelint.file_utils import (
  File "/usr/lib/python3/dist-packages/ansiblelint/file_utils.py", line 15, in <module>
    import wcmatch.pathlib
  File "/usr/lib/python3/dist-packages/wcmatch/pathlib.py", line 218, in <module>
    class PurePosixPath(PurePath):
  File "/usr/lib/python3/dist-packages/wcmatch/pathlib.py", line 221, in PurePosixPath
    _flavour = pathlib._posix_flavour  # type: ignore[attr-defined]
               ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'pathlib' has no attribute '_posix_flavour'
```
(Ubuntu 24.04, Python 3.12, installed from apt)

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```ansible-lint --version```

##### Desired Behavior

ansible-lint not to crash but do the linting things.

##### Actual Behavior

ansible-lint crashes

**Comment by ssbarnea** [comment]
This is not a linter bug, it is caused by the wrong distribution. Canonical already had a bug reported about and they did not fix it yet.

Please do take the due diligence when filing bug, there is an already closed one here and also one on launchpad.

**Comment by Koenvh1** [comment]
@ssbarnea I searched for other issues here first, but the GitHub search did not turn up any results, hence I opened the issue. Now manually going through the list of closed issues I assume you refer to #4123 ?

**Comment by Qalthos** [comment]
That is the closed bug, and https://bugs.launchpad.net/ubuntu/+source/python-wcmatch/+bug/2060649 is the one opened against Ubuntu. Please follow the latter for updates as there isn't anything we can do about it. Thanks

**Comment by The-Alley** [comment]
Hi,

I am not sure if this helps or not but anyways:

The problem is with Ubuntu's `python3-wcmatch` version 8.4.2 which is broken and gives `AttributeError: module 'pathlib' has no attribute '_posix_flavour'` error. The latest version of the actual python package `wcmatch` is 8.5.2 and it works. So AFAIK there are two solutions here:
1. Wait for future releases of `python3-wcmatch` since its developers are already aware of the issue.
2. On Ubuntu, create a Python virtual env, install latest version install `ansible-lint` using pip since now it requires newer compatible versions of `wcmatch`.
3. If you really want to have `ansible-lint` on Ubuntu's global Python without creating envs, then you have to use pip to  **forcibly** install  latest version of `wcmatch`  using `python3 -m pip install --break-system-packages wcmatch` **WHCIH IS HIGHLY NON RECOMMENDED**. 

TL;DR: If you get that error, install `ansible-lint` using pip not apt for now.

**Comment by TheDavidFactor** [comment]
I chose option 3, because the system package is already broken, and you need to add `--upgrade` to upgrade the installed `wcmatch`
